src/components/input-field/input-field.tsx

```diff
@@ -292,32 +295,35 @@ export class InputField {
         properties.readonly = this.readonly;
         properties.disabled = this.disabled || this.readonly;
 
+        if (this.hasHelperText()) {
+            properties['aria-controls'] = helperTextId;
+            properties['aria-describedby'] = helperTextId;
+        }
+
         const labelClassList = {
             'mdc-floating-label': true,
             'mdc-floating-label--float-above':
                 !!this.value || this.isFocused || this.readonly,
         };
 
         return [
-            <div class={this.getContainerClassList()}>
+            <label class={this.getContainerClassList()}>
```

Comment:
Why is this `<label` instead of `<div`? On MD's site, it's rendered as a div.
![image](https://user-images.githubusercontent.com/35954987/125779505-457685dd-38fa-4280-9dbb-9a2cd82aeaa4.png)

Maybe this doesn't affect the styles. But it feels semantically wrong to call it label. What do you say? Can I change it to div @adrianschmidt?

Reply:
I don't think it's semantically wrong. It's semantically correct to write either
```html
<label for="blah">Blah blah</label>
<input id="blah" />
```
or
```html
<label>
    Blah blah
	<input />
</label>
```
and the code we use is equivalent to the latter example, just with a lot more gunk…

The code I've used comes straight from MDC's docs for v11.0.0: https://github.com/material-components/material-components-web/tree/v11.0.0/packages/mdc-textfield#outlined-text-example

Reply:
Huh 🤔. OK. Strange that they have div on their documentations site.

Reply:
I think material.io is more about documenting the design ideas, rather than the specific implementation. They might not be keeping their MDC version up to date. They also don't specify which version they are using, so it's pretty clear to me that the intention isn't for those docs to be technical documentation about any one specific implementation.